esp-hal-common/src/rtc_cntl/rtc/esp32c6.rs

```diff
@@ -0,0 +1,697 @@
+    fn ets_delay_us(us: u32);
+}
+
+#[allow(unused)]
+#[derive(Debug, Clone, Copy)]
+/// RTC SLOW_CLK frequency values
+pub(crate) enum RtcFastClock {
+    /// Select RC_FAST_CLK as RTC_FAST_CLK source
+    RtcFastClockRcFast = 0,
+    /// Select XTAL_D2_CLK as RTC_FAST_CLK source
+    RtcFastClockXtalD2 = 1,
+}
+
+impl Clock for RtcFastClock {
+    fn frequency(&self) -> HertzU32 {
+        match self {
+            RtcFastClock::RtcFastClockXtalD2 => HertzU32::Hz(40_000_000 / 2), /* TODO: Is the value correct? */
+            RtcFastClock::RtcFastClockRcFast => HertzU32::Hz(17_500_000),
+        }
+    }
+}
+
+#[derive(Debug, Clone, Copy, PartialEq)]
+/// RTC SLOW_CLK frequency values
+pub(crate) enum RtcSlowClock {
+    /// Select RC_SLOW_CLK as RTC_SLOW_CLK source
+    RtcSlowClockRcSlow  = 0,
+    /// Select XTAL32K_CLK as RTC_SLOW_CLK source
+    RtcSlowClock32kXtal = 1,
+    /// Select RC32K_CLK as RTC_SLOW_CLK source
+    RtcSlowClock32kRc   = 2,
+    /// Select OSC_SLOW_CLK (external slow clock) as RTC_SLOW_CLK source
+    RtcSlowOscSlow      = 3,
+}
+
+impl Clock for RtcSlowClock {
+    fn frequency(&self) -> HertzU32 {
+        match self {
+            RtcSlowClock::RtcSlowClockRcSlow => HertzU32::Hz(136_000),
+            RtcSlowClock::RtcSlowClock32kXtal => HertzU32::Hz(32_768),
+            RtcSlowClock::RtcSlowClock32kRc => HertzU32::Hz(32_768),
+            RtcSlowClock::RtcSlowOscSlow => HertzU32::Hz(32_768),
+        }
+    }
+}
+
+#[derive(Debug, Clone, Copy, PartialEq)]
+/// Clock source to be calibrated using rtc_clk_cal function
+pub(crate) enum RtcCalSel {
+    /// Currently selected RTC SLOW_CLK
+    RtcCalRtcMux     = -1,
+    /// Internal 150kHz RC oscillator
+    RtcCalRcSlow     = 0,
+    /// External 32kHz XTAL, as one type of 32k clock
+    RtcCal32kXtal    = 1,
+    /// Internal 32kHz RC oscillator, as one type of 32k clock
+    RtcCal32kRc      = 2,
+    /// External slow clock signal input by lp_pad_gpio0, as one type of 32k
+    /// clock
+    RtcCal32kOscSlow = 3,
+    /// Internal 20MHz RC oscillator
+    RtcCalRcFast,
+}
+
+#[derive(Clone)]
+pub(crate) enum RtcCaliClkSel {
+    CaliClkRcSlow = 0,
+    CaliClkRcFast = 1,
+    CaliClk32k    = 2,
+}
+/// RTC Watchdog Timer
+pub struct RtcClock;
+
+/// RTC Watchdog Timer driver
+impl RtcClock {
+    const CAL_FRACT: u32 = 19;
+
+    /// Enable or disable 8 MHz internal oscillator
+    fn enable_8m(clk_8m_en: bool, _d256_en: bool) {
+        let pmu = unsafe { &*PMU::PTR };
+
+        if clk_8m_en {
+            pmu.hp_sleep_lp_ck_power
+                .modify(|_, w| w.hp_sleep_xpd_fosc_clk().set_bit());
+
+            unsafe { ets_delay_us(50) };
+        } else {
+            pmu.hp_sleep_lp_ck_power
+                .modify(|_, w| w.hp_sleep_xpd_fosc_clk().clear_bit());
+        }
+    }
+
+    /// Get main XTAL frequency
+    /// This is the value stored in RTC register RTC_XTAL_FREQ_REG by the
+    /// bootloader, as passed to rtc_clk_init function.
```

Comment:
seems like the 2nd stage bootloader just reads the value so it should be the ROM bootloader but probably we should just double check it

Reply:
Seems I was wrong - in direct-boot the read is invalid and we fall back to 40MHz which is anyway the only xtal frequency we currently support for C6

Since we need to know the xtal frequency anyway at build time (i.e. we have special features for 26MHz xtals for ESP32-C2 and ESP32 - for all others we only support 40MHz) we probably can just always use that known value

But probably nothing that needs to get addressed in this PR since it's like that for all targets and not really causing problems AFAICS